pkg/controllers/dynakube/activegate/internal/authtoken/reconciler.go

```diff
@@ -41,19 +44,6 @@ func NewReconciler(clt client.Client, apiReader client.Reader, dynakube *dynatra
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	if !r.dynakube.NeedsActiveGate() {
```

Comment:
why was this removed?

I know we check in the AG reconciler already, but that is not how it should be done:
- Each `Reconciler` should "own" the things they create, so the AG reconciler should just call the `AuthToken` reconciler, that will clean up OR create what is necessary.
  - Otherwise, to do cleanup, each `Reconciler` would need a `CleanUp` function that is called by the original "parent" calling the `Reconciler`.
    - This unnecessarily spreads the logic between packages (makes testing a bit more annoying)

    
The line where you added the cleanup of the new condition has a comment: `// TODO: move cleanup to ActiveGate reconciler`, that now would need another: `// TODO: move cleanup to AuthToken reconciler` 😅 


IMO how a `Reconile` func should look like: (not everything needs to be sooo abstracted, it was just easier to write here 😅 )

```go
func (r *Reconciler) Reconcile(ctx context.Context) error {
	if !r.needed() { 
		return r.cleanUp()
	}
	return r.doStuff()
}
```

Reply:
I wasn't sure the meaning of `// TODO: move cleanup to ActiveGate reconciler` so I moved cleanup to AG reconciler :smile: .

Fixed. The auth-token condition is removed by authtoken reconciler.